File: TenMinsJava/day100.java
// press "c" to add new column to the csv
package TenMinsJava;

import java.io.File;
import java.io.FileWriter;
import java.util.ArrayList;
import java.util.Scanner;

public class day100 {
    public static void addColumn(String columnName) {
        try {
            File file = new File("D:/coding30minsAday/textFiles/JTrialCsv.csv");
            Scanner fScan = new Scanner(file);
            ArrayList<String> lines = new ArrayList<>();

            lines.add(fScan.nextLine() + "," + columnName);
            while (fScan.hasNextLine()) {
                lines.add(fScan.nextLine());
            }
            fScan.close();

            // rewrite the csv
            FileWriter w = new FileWriter(file);
            for (String string : lines) {
                w.write(string + "\n");
            }
            w.close();
        } catch (Exception e) {
            System.err.println("Error: " + e);
        }
    }
    class Main {
        public static void main(String[] args) {
            Scanner scan = new Scanner(System.in);
            System.out.println("Press c to add new column");
            String choice = scan.nextLine();
            if (choice.trim().toLowerCase().equals("c")) {
                System.out.println("Enter column name");
                String search = scan.nextLine().trim();
                addColumn(search);
            }
            scan.close();
        }
    }
}

File: TenMinsPy/day100.py
# press "c" to add new column to the csv

def addColumn(columnName, file_path="D:/coding30minsAday/textFiles/JTrialCsv.csv"):
    try:
        # Step 1: Read all lines
        with open(file_path, "r") as f:
            lines = f.readlines()
            
        lines[0] = lines[0].strip() + "," + columnName + "\n"
        for i in range(1, len(lines)):
            lines[i] = lines[i].strip() + ",\n"

        # Step 3: Rewrite the CSV
        with open(file_path, "w") as f:
            f.writelines(lines)

    except Exception as e:
        print("Error:", e)

def ask():
    choice = input("enter c to add new column name\n").strip()
    if choice.lower() == "c":
        row_id = input("enter columnName")
        addColumn(row_id)

ask()